static/js/browser2/observation_chart.tsx

```diff
@@ -0,0 +1,273 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component for displaying a chart for a single source series for a place stat var.
+ */
+
+import React from "react";
+import _ from "lodash";
+import * as d3 from "d3";
+import axios from "axios";
+
+// Chart size
+const WIDTH = 500;
+const HEIGHT = 250;
+
+// Chart margin.
+const TOP = 50;
+const RIGHT = 230;
+const BOTTOM = 50;
+const LEFT = 100;
+
+const LINE_STROKE_COLOR = "rgb(31, 119, 180)";
+const URI_PREFIX = "/browser2/";
+const TOOLTIP_ID = "tooltip";
+
+interface ObservationChartPropType {
+  // TODO (chejennifer): get rid of the any type
+  sourceSeries: any;
+  idx: number;
+  statVarId: string;
+  placeDcid: string;
+}
+
+export class ObservationChart extends React.Component<
+  ObservationChartPropType
+> {
+  constructor(props: ObservationChartPropType) {
+    super(props);
+  }
+
+  componentDidMount(): void {
+    this.plot();
+  }
+
+  render(): JSX.Element {
+    return (
+      <div className="card">
+        <div id={this.props.statVarId} className="chart-title">
+          <div>
+            {this.props.sourceSeries.measurementMethod
```

Comment:
you can try using "mmethod", "obsPeriod", "sfactor" to make the code shorter. We have been widely using these in g3

Reply:
Spelling it out is reasonable though, so since it's written already, I'd prefer to keep it for readability.

Reply:
I'm accessing the object returned from get_stats_all, so I don't think I can just use shorter names without doing some kind of processing?

Reply:
I see, then using the full name is ok